### DISCUSSION TITLE
Automate the process of upgrading the library included in the addon

### DISCUSSION THREAD

**Discussion by florian-h05** [openhab/openhab-js#242] [JS Maintainers Discussions]
As already discussed in a previous discussion, automatically publishing new versions from the repo is no easy job because we have our CHANGELOG and so on.

But I see some other way to improve our publishing workflow: The process of upgrading the version included in the add-on.

On each new version published, we basically change the version numbers in the add-ons `pom.xml` and replace everything in the addons README that is under a comment (`<!-- Paste the copied docs from openhab-js under this comment. --> `) with what‘s under a specific comment (`<!-- Copy everything from here to update the JS Scripting documentation. --> `) in the library‘s README.

We would need a CI action that is triggered when a new tag `v*` is pushed and that action needs the files mentioned above and open a PR to the addons repo.

@rkoshak 
Referencing https://community.openhab.org/t/pushover-notifications-from-javascript-rules/130697/15.

**Comment by florian-h05** [comment]
We could also use `latest` as the version from npm, then we would just need to include the part of the addon docs maintained at openhab-js repo using some Jekyll magic.

@Confectrician
Is it possible to include a markdown file from the openhab-js repo in the README of the JS Scripting addon?

**Comment by Confectrician** [comment]
Do you speak about dynamically including it during the docs build?
Can you give me some more information what you want to achieve?

**Comment by florian-h05** [comment]
> Do you speak about dynamically including it during the docs build?

Yes.

> Can you give me some more information what you want to achieve?

The largest part of the JS Scripting docs is in fact maintained at the openhab-js repo and has to be synced with the repo every time we release a new version of openhab-js and include it in the addon.

I would like to dynamically include the part of the addon docs that is maintainer here in the addon docs during build time of the docs.
I can move that part to an extra markdown file, so there is basically a markdown file in the openhab-js repo that has to be dynamically included in the JS Scripting add-on docs during the docs build.
If possible, the docs build should use this markdown file from the latest tag added to the openhab-js repo; so that the docs are in sync with the version published to npm and included in the addon.

**Comment by Confectrician** [comment]
This should be possible in theory.
We are doing this in a quite similar way for alexa and openhabian based on their own repositories.
Those repositories are even able to trigger an update PR via GitHub automation on their own, which would also be interesting for this usescase, when the library itself gets a new tag.

But if i get it right we would then need to build both websites. The stable and latest docs.
Correct?

I still have to see, how we can change one addon-doc file in an easy to maintain way and will make some thoughts about this.

**Comment by florian-h05** [comment]
This sound good.

> But if i get it right we would then need to build both websites. The stable and latest docs.
Correct?

Yes. Stable need to use the docs matching the included version of the library, while latest needs to use the docs of the latest release of the library.
